include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -79,18 +79,26 @@ struct __make_references
 template <typename... _Types>
 class zip_forward_iterator
 {
+    template <typename... _Ts>
+    using __tuple_t =
+#if _ONEDPL_CAN_USE_STD_TUPLE_PROXY_ITERATOR
+        ::std::tuple<_Ts...>;
+#else
+        oneapi::dpl::__internal::tuple<_Ts...>;
+#endif
+
     static const ::std::size_t __num_types = sizeof...(_Types);
-    typedef typename oneapi::dpl::__internal::tuple<_Types...> __it_types;
+    typedef __tuple_t<_Types...> __it_types;
```

Comment:
Since `__tuple_t` is not qualified, I believe it is incorrect to have `typename` here as there is no ambiguity it resolves. Doing so results in the following compilation error:
```
. . . oneDPL/include/oneapi/dpl/pstl/iterator_impl.h:91:22: error: expected nested-name-specifier
   91 |     typedef typename __tuple_t<_Types...> __it_types;
```

Reply:
aha, yes.  Thanks for the explanation.  You are correct.